backend/database-router.js

```diff
@@ -23,9 +23,9 @@ export default class RouterBackend extends DatabaseBackend implements IDatabaseB
   backends: { [string]: DatabaseBackend }
   config: Config
 
-  constructor (options: Object = {}) {
+  constructor (options: { config?: Config } = {}) {
     super()
-    this.config = options.config ?? null
+    if (options.config) this.config = options.config
```

Comment:
Minor comment: I think this _still_ doesn't address the type issue. You're saying that `config` has type `Config` (in line 24), but instead it has type `?Config` (or `Config | typeof undefined`, before it was `Config`).

I see that if you call `init` as you should, then `config` will then have the correct type. I think the types as they're currently defined are inaccurate, but I'm not sure if this is worth pursuing further.

Reply:
@corrideat when this PR is merged, can you fix this when you merge `master` into your PR that I'm currently reviewing?

Reply:
@corrideat Yeah instance properties don't have their correct types yet when exiting the constructor call. They have to wait until `init()` has returned successfully. For example, the `backends` property isn't set in the constructor either which means `this.backends` is initially `undefined`.